actionview/lib/action_view/helpers/tags/base.rb

```diff
@@ -14,7 +14,7 @@ def initialize(object_name, method_name, template_object, options = {})
           @object_name.sub!(/\[\]$/,"") || @object_name.sub!(/\[\]\]$/,"]")
           @object = retrieve_object(options.delete(:object))
           @options = options
-          @auto_index = retrieve_autoindex(Regexp.last_match.pre_match) if Regexp.last_match
+          @auto_index = Regexp.last_match ? retrieve_autoindex(Regexp.last_match.pre_match) : nil
```

Comment:
It is better to define `@auto_index` variable all the time other than check if it is defined or not.